File: frontend/src/types/category.ts
export interface Category {
  id: number;
  name: string;
  description?: string;
  parent_id?: number;
}

export interface CategoryCreate {
  name: string;
  description?: string;
  parent_id?: number;
}

export interface CategoryUpdate {
  name?: string;
  description?: string;
  parent_id?: number;
}

File: backend/app/schemas/category.py
from pydantic import BaseModel, ConfigDict
from typing import Optional


# Base schema
class CategoryBase(BaseModel):
    name: str
    description: Optional[str] = None
    parent_id: Optional[int] = None


# Schema for category creation
class CategoryCreate(CategoryBase):
    pass


# Schema for category update
class CategoryUpdate(BaseModel):
    name: Optional[str] = None
    description: Optional[str] = None
    parent_id: Optional[int] = None


# Schema for category in DB
class Category(CategoryBase):
    id: int

    model_config = ConfigDict(from_attributes=True)
